src/Microsoft.Identity.Client/User.cs

```diff
@@ -47,37 +46,34 @@ internal User(User other)
             IdentityProvider = other.IdentityProvider;
         }
 
+        public User(string identifier, string displayableId, string name, string identityProvider)
+        {
+            if (string.IsNullOrWhiteSpace(identifier))
+            {
+                throw new ArgumentNullException(nameof(identifier));
+            }
+
+            DisplayableId = displayableId;
+            Name = name;
+            IdentityProvider = identityProvider;
```

Comment:
Users need to be able to test that their apps can use the returned _User_ objects correctly, so either they need to be able to create _User_ objects or there needs to be an _IUser_ interface. As this is now a data-only class with only simple write-once fields it didn't seem worth creating an interface. However, I'll create one if you prefer.